lib/engine/game/g_18_eus/game.rb

```diff
@@ -472,6 +485,132 @@ def company_status_str(company)
+        end
+
+        def bny
+          @bny ||= @corporations.find { |c| c.type == :bank }
+        end
+
+        def setup_bny
+          stock_market.set_par(bny, stock_market.par_prices.find { |pp| pp.price == 80 })
+          bny.ipoed = true
+          bny.owner = @share_pool
+          setup_loans
+        end
+
+        def remove_subsidies; end
+
+        def setup_loans
+          @loans =
+            case @players.size
+            when 3
+              [
+                { 'stock_movement' => :diagonal, 'multipliers' => [0.5, 1, 1, 1.5, 1.5, 2] },
+                { 'stock_movement' => :straight, 'multipliers' => [2, 2, 2, 2.5, 2.5, 2.5, 3, 3] },
+                { 'stock_movement' => :diagonal_and_straight, 'multipliers' => [3, 3, 3.5, 3.5, 3.5, 3.5, 4, 4] },
+                { 'stock_movement' => :diagonal_and_straight, 'multipliers' => [4, 4, 5, 5, 5, 5, 5, 5] },
+              ]
+            when 4
+              [
+                { 'stock_movement' => :diagonal, 'multipliers' => [0.5, 0.5, 1, 1, 1.5, 1.5, 2] },
+                { 'stock_movement' => :straight, 'multipliers' => [2, 2, 2.5, 2.5, 2.5, 3, 3] },
+                { 'stock_movement' => :diagonal_and_straight, 'multipliers' => [3, 3, 3, 3.5, 3.5, 3.5, 3.5, 3.5] },
+                { 'stock_movement' => :diagonal_and_straight, 'multipliers' => [3.5, 3.5, 4, 4, 4, 4, 4, 4, 4] },
+                { 'stock_movement' => :diagonal_and_straight, 'multipliers' => [5, 5, 5, 5, 5, 5, 5, 5, 5] },
+              ]
+            when 5
+              [
+                { 'stock_movement' => :diagonal, 'multipliers' => [0.5, 0.5, 1, 1, 1.5, 1.5, 1.5, 2] },
+                { 'stock_movement' => :straight, 'multipliers' => [2, 2, 2, 2.5, 2.5, 2.5, 3, 3] },
+                { 'stock_movement' => :diagonal_and_straight, 'multipliers' => [3, 3, 3, 3, 3.5, 3.5, 3.5, 3.5, 3.5, 4] },
+                { 'stock_movement' => :diagonal_and_straight, 'multipliers' => [4, 4, 4, 4, 4, 4, 4, 5, 5, 5] },
+                { 'stock_movement' => :diagonal_and_straight, 'multipliers' => [5, 5, 5, 5, 5, 5, 5, 5, 5, 5] },
+              ]
+            end
+          @loans_taken = 0
+        end
+
+        def loan_chart
+          loan_chart = []
+          @loans.each do |row|
+            loan_chart << ([translate_loan_movement(row['stock_movement'])] + row['multipliers']).flatten
+          end
+          loan_chart
+        end
+
+        def translate_loan_movement(movement)
+          case movement
+          when :diagonal
+            '↗'
+          when :straight
+            '→'
+          when :diagonal_and_straight
+            '→↗'
+          end
+        end
+
+        def current_loan_values
+          return [:none, 0] if @loans_taken.zero?
+
+          taken = @loans_taken
+          @loans.each do |row|
+            return [row['stock_movement'], row['multipliers'][taken - 1]] if taken <= row['multipliers'].size
+
+            taken -= row['multipliers'].size
+          end
```

Comment:
Came up with another way - instead of recalculating "row" and "col" all the time, built a map during setup so that I can just get the row and col from the current number of taken loans.

Reply:
Also removed this method because never needed to get stock_movement and multiplier at the same time - so now have two methods, one for each.